flathunter/crawler/immowelt.py

```diff
@@ -49,61 +49,68 @@ def get_expose_details(self, expose):
     def extract_data(self, soup: BeautifulSoup):
         """Extracts all exposes from a provided Soup object"""
         entries = []
-        soup_res = soup.find("main")
+        soup_res = soup
         if not isinstance(soup_res, Tag):
             return []
 
-        title_elements = soup_res.find_all("h2")
-        expose_ids = soup_res.find_all("a", id=True)
+        advertisements = soup_res.find_all("div", attrs={"class": "css-79elbk"})
+        for adv in advertisements:
+            try:
+                title = adv.find("div", {"class": "css-1cbj9xw"}).text
+            except:
+                title = ""
 
-        for idx, title_el in enumerate(title_elements):
             try:
-                price = expose_ids[idx].find(
-                    "div", attrs={"data-test": "price"}).text
-            except IndexError:
+                price = adv.find(
+                    "div", attrs={"data-testid": "cardmfe-price-testid"}).text
+            except:
                 price = ""
 
             try:
-                size = expose_ids[idx].find(
-                    "div", attrs={"data-test": "area"}).text
+                descriptions = adv.find("div", attrs={"data-testid": "cardmfe-keyfacts-testid"}).children
+                descriptions = [result.text for result in descriptions]
+            except:
+                descriptions = []
+            
+            size = list(filter(lambda x: "m²" in x, descriptions))
+            try:
+                size = size[0]
             except IndexError:
                 size = ""
 
+            rooms = list(filter(lambda x: "Zimmer" in x, descriptions))
             try:
-                rooms = expose_ids[idx].find(
-                    "div", attrs={"data-test": "rooms"}).text.replace(" Zi.", "")
+                rooms = rooms[0]
             except IndexError:
                 rooms = ""
 
+            id_element = adv.find("a")
             try:
-                url = expose_ids[idx].get("href")
+                url = "https://immowelt.de" + id_element.get("href")
```

Comment:
Some comment on this after some more testing from me:
The extracted "href" already contained the complete URL, so adding "https://immowelt.de" in front created some "doubled" URL, which was especially annoying when getting the information from the Telegram bot, since you could click on the wrong part of the link.

I updated my code with this one, just in case (for whatever) reason, the first part of the URL isn't in the "href".
```suggestion
                url = id_element.get("href")
                if "https" not in url:
                    url = "https://immowelt.de/" + url
```

Reply:
Yes, they changed that after a month from my change. I changed it on my branch as well. Good eye